File: be/src/main/java/sg/flow/repositories/account/AccountRepository.java
package sg.flow.repositories.account;

import java.util.List;

import sg.flow.entities.Account;
import sg.flow.models.account.AccountWithTransactionHistory;
import sg.flow.models.account.BriefAccount;
import sg.flow.repositories.Repository;

public interface AccountRepository extends Repository<Account, Long> {

    List<BriefAccount> findBriefAccountsOfUser(Integer userId);

    List<AccountWithTransactionHistory> findAccountWithTransactionHistorysOfUser(Integer userId);

}

File: be/src/main/kotlin/sg/flow/repositories/account/AccountRepository.kt
package sg.flow.repositories.account

import sg.flow.entities.Account
import sg.flow.entities.Card
import sg.flow.models.account.AccountWithTransactionHistory
import sg.flow.models.account.BriefAccount


import sg.flow.repositories.Repository

interface AccountRepository : Repository<Account, Long> {
    suspend fun findAccountsOfUser(userId: Int): List<BriefAccount>
    suspend fun findAccountWithTransactionHistorysOfUser(userId: Int): List<AccountWithTransactionHistory>
    suspend fun findAccountByFinverseAccountId(finverseAccountId: String): Account?
    suspend fun findCardAccounts(userId: Int): List<Card>
} 